open-metadata-implementation/adapters/open-connectors/integration-connectors/elasticsearch-integration-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/elasticsearch/ElasticsearchIntegrationConnector.java

```diff
@@ -0,0 +1,103 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+
+package org.odpi.openmetadata.adapters.connectors.integration.elasticsearch;
+
+
+import org.odpi.openmetadata.adapters.connectors.integration.elasticsearch.ffdc.ElasticsearchIntegrationConnectorAuditCode;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.ConnectorCheckedException;
+import org.odpi.openmetadata.frameworks.connectors.properties.ConnectionProperties;
+import org.odpi.openmetadata.integrationservices.search.connector.SearchIntegratorConnector;
+import org.odpi.openmetadata.integrationservices.search.connector.SearchIntegratorContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+
+/**
+ * ElasticsearchIntegrationConnector provides common methods for the connector in this module.
+ */
+public class ElasticsearchIntegrationConnector extends SearchIntegratorConnector
+{
+    private static final Logger log = LoggerFactory.getLogger(ElasticsearchIntegrationConnector.class);
+
+    private String targetRootURL = null;
+
+    private SearchIntegratorContext myContext = null;
+
+    /**
+     * Initialize the connector.
+     *
+     * @param connectorInstanceId - unique id for the connector instance - useful for messages etc
+     * @param connectionProperties - POJO for the configuration used to create the connector.
+     */
+    @Override
+    public void initialize(String connectorInstanceId, ConnectionProperties connectionProperties)
+    {
+        super.initialize(connectorInstanceId, connectionProperties);
+
+        org.odpi.openmetadata.frameworks.connectors.properties.EndpointProperties  endpoint = connectionProperties.getEndpoint();
+
+        if (endpoint != null)
```

Comment:
if the end point is null should it fail to initialise ? Or is it ok to proceed with a null endpoint.

Reply:
this is a pattern that I saw is used in the integration connectors, the initialisation continues even if the endpoint properties are null.